### PR TITLE
Port katello_product to cement

### PR DESCRIPTION
**Pull request by mdellweg**
* add test data

affects: #37 

Introduces the ability to delete Products.

### CODE REVIEW

**Comment by akofink** [comment]
Ping me when ~WIP~

**Comment by mdellweg** [comment]
@akofink It is functional right now, but i want to have a look whether i can easily add the gpg-key and sync-plan options. If not, they might go in another PR.

**Comment by jdshewey** [comment]
@mdellweg - Not sure if this is helpful, but you might be able to get some code for adding gpg keys and sync plans from [here](https://github.com/jdshewey/salt-formula-katello/blob/master/_states/katello.py)

**Comment by mdellweg** [comment]
@jdshewey Thanks, that shows at least, that Katello's apiv2 is capable. But we kind of bound to use nailgun. I will look into it.

File: test/demo/product.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  - name: 'Edit the product'
+    katello_product:
+      username: "{{ foreman_username }}"
+      password: "{{ foreman_password }}"
+      server_url: "{{ foreman_server_url }}"
+      verify_ssl: "{{ foreman_verify_ssl }}"
+      name: "Katello's finest"
+      organization: 'Default Organization'
+      description: "Fancy assortment of well matured cask strength repositories"
+      state: present
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply test/demo/product.yml]: It doesn't matter. Both of them create a new object if there is none. Only 'present' updates if there are differences.

**Comment by mdellweg** [comment]
Sadly, there are neither gpg_key nor sync_plan options in Nailgun. So i will release this PR from WIP. ;)

File: test/demo/product.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (akofink) [reply test/demo/product.yml]: Oh, hmm... I know we've had this discussion before about the state conventions in this repo. Do you happen to have a link?

[the remark this answers is not in the document]
Comment (mdellweg) [reply test/demo/product.yml]: https://github.com/theforeman/foreman-ansible-modules/issues/30

**Pull request merged**